src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -466,6 +469,71 @@ RequestContext<object> requestContext
             await requestContext.SendResult(result.First().ToString());
         }
 
+        private async Task HandleGetAllCommandsRequest(
+            object param,
+            RequestContext<object> requestContext)
+        {
+            var psCommand = new PSCommand();
+            psCommand.AddScript("Get-Command -CommandType Function, Cmdlet, ExternalScript | Select-Object Name,ModuleName | Sort-Object Name");
+            Logger.Write(LogLevel.Verbose, $"Calling {psCommand.Commands}");
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+            var result = await this.editorSession.PowerShellContext.ExecuteCommand<PSObject>(psCommand);
+            stopwatch.Stop();
+            Logger.Write(LogLevel.Verbose, $"Returned from Get-Command with {result.Count()} results in {stopwatch.ElapsedMilliseconds}ms");
+            var commandList = new List<PSAllCommandsMessage>();
+
+            if (result != null)
+            {
+                Logger.Write(LogLevel.Verbose, "Starting conversion of results");
+                foreach (dynamic c in result)
+                {
+                    commandList.Add(new PSAllCommandsMessage
+                    {
+                        Name = c.Name,
+                        ModuleName = c.ModuleName,
+                    });
+                }
+                Logger.Write(LogLevel.Verbose, "Finished conversion of results");
+            }
+
+            await requestContext.SendResult(commandList);
+        }
+
+        private async Task HandleGetCommandRequest(
+            string param,
+            RequestContext<object> requestContext)
+        {
+            var psCommand = new PSCommand();
```

Comment:
For both of these methods we're using a similar pattern twice:
1. Compose PowerShell command
2. Log call
3. Start stopwatch
4. Run command
5. Stop stopwatch
6. Log call and performance

I feel like steps 2-6 should be done for every PowerShell command (i.e. should be implicit in the `ExecuteCommand` call), but certainly here should be turned into their own function

Reply:
This was mostly put in while I was troubleshooting a performance issue where it was taking 4 - 5 seconds to come back where the command in normal powershell was under a second. Turns out the time was being consumed by sending 800,000 lines of json (hence we now get all commands only and have a separate message to get all of the command information).

To avoid refactoring it out later, I've removed this pattern in favor of adding it as a separate PR to ExecuteCommand.